src/display/process/allProcs.go

```diff
@@ -35,77 +36,67 @@ import (
 var runAllProc = true
 var helpVisible = false
 
-func getData(procs []*proc.Process) []string {
-	var data []string
-	for _, info := range procs {
-		exe, err := info.Exe()
+func getData(procs []*proc.Process) [][]string {
+	procData := [][]string{}
+	for _, p := range procs {
+		// Get command
+		cmd := ""
+		exe, err := p.Exe()
 		if err == nil {
-			temp := strconv.Itoa(int(info.Pid))
+			cmds := strings.Split(exe, "/")
+			cmd = cmds[len(cmds)-1]
 
-			for i := 0; i < 12-len(strconv.Itoa(int(info.Pid))); i++ {
-				temp = temp + " "
-			}
-
-			commands := strings.Split(exe, "/")
-			command := commands[len(commands)-1]
-
-			if len(command) > 40 {
-				temp = temp + "[" + command[:40] + "](fg:green)" + strings.Repeat(" ", 41-len(command))
-			} else {
-				temp = temp + "[" + command + "](fg:green)" + strings.Repeat(" ", 41-len(command))
-			}
-
-			tempCPU, err := info.CPUPercent()
-			cpuPercent := ""
+			// Get CPU
+			cpu := ""
+			cpuPercent, err := p.CPUPercent()
 			if err == nil {
-				cpuPercent = fmt.Sprintf("%.2f%s", tempCPU, "%")
-				temp = temp + cpuPercent
+				cpu = fmt.Sprintf("%.2f%%", cpuPercent)
 			}
-			temp = temp + strings.Repeat(" ", 11-len(cpuPercent))
 
-			tempMem, err := info.MemoryPercent()
-			memPercent := ""
+			// Get Mem
+			mem := ""
+			memPercent, err := p.MemoryPercent()
 			if err == nil {
-				memPercent = fmt.Sprintf("%.2f%s", tempMem, "%")
-				temp = temp + memPercent
+				mem = fmt.Sprintf("%.2f%%", memPercent)
 			}
-			temp = temp + strings.Repeat(" ", 11-len(memPercent))
 
-			status, err := info.Status()
-			if err == nil {
-				temp = temp + status
-			}
-			temp = temp + strings.Repeat(" ", 9-len(status))
+			// Get Status
+			status, _ := p.Status()
 
-			fg, err := info.Foreground()
-			if err == nil {
-				if fg {
-					temp = temp + "True"
-					temp = temp + strings.Repeat(" ", 9)
-				} else {
-					temp = temp + "False"
-					temp = temp + strings.Repeat(" ", 8)
-				}
-			}
+			// Get Foreground
+			fg, _ := p.Foreground()
 
-			ctime, err := info.CreateTime()
-			createTime := ""
+			// Get Creation time
+			t, err := p.CreateTime()
+			ctime := ""
 			if err == nil {
-				createTime := utils.GetDateFromUnix(ctime)
-				temp = temp + createTime
+				ctime = utils.GetDateFromUnix(t)
 			}
-			temp = temp + strings.Repeat(" ", 9-len(createTime))
 
-			threads, err := info.NumThreads()
-			if err == nil {
-				threadCount := strconv.FormatInt(int64(threads), 10)
-				temp = temp + threadCount
+			// Get Thread Count
+			tc, _ := p.NumThreads()
+
+			// Aggregate row
+			r := []string{
+				fmt.Sprintf("%d", p.Pid),
+				cmd,
+				cpu,
+				mem,
+				status,
+				fmt.Sprintf("%t", fg),
+				ctime,
+				fmt.Sprintf("%d", tc),
 			}
-
-			data = append(data, temp)
+			procData = append(procData, r)
 		}
 	}
-	return data
+
+	// Sort on Timestamp
+	sort.Slice(procData, func(i, j int) bool {
```

Comment:
I'm going to open an issue, revert this for now if possible.

Reply:
I'd prefer to leave this as default sorting. Else the list changes often and gets messy to navigate. With the issue open we can extend this based on user choice. What do you think?

Reply:
Not sure if a default sort should be there at all, its unnecessary overhead, if sorting is desired, the to-be keybindings could be used imo

Reply:
The proc list changes on every refresh, this makes it hard to focus on the proc being read. Plus the sort isn't very heavy in itself so I feel it should stay (if nothing else at least as placeholder).

Reply:
> if nothing else at least as placeholder

Placeholder to what?

> Plus the sort isn't very heavy in itself

On what basis?

Reply:
> Placeholder to what?
To later implemented sorting

> On what basis?
CPU consumed is fairly high for proc updates, the sort overhead is almost negligible in comparison.

Reply:
> The proc list changes on every refresh, this makes it hard to focus on the proc being read.

For this, I'd assume someone would use `grofer proc -p <PID>`. `grofer proc` is just to give an overall view of the system processes IMO, and then if you'd like something like the most CPU intensive process or something, you'd sort. And to get the PID based on the comamnd, you could `pause`, I believe we have a keybinding for that. 

And besides, `grofer proc` is one of our most CPU intensive commands, better to keep the default case as light as possible.

Reply:
Fair enough, no sort by default.